controllers/periodic_healthcheck.go

```diff
@@ -130,44 +131,76 @@ func (r *EtcdadmClusterReconciler) startHealthCheck(ctx context.Context, etcdadm
 
 func (r *EtcdadmClusterReconciler) periodicEtcdMembersHealthCheck(ctx context.Context, cluster *clusterv1.Cluster, etcdCluster *etcdv1.EtcdadmCluster, etcdadmClusterMapper map[types.UID]etcdadmClusterMemberHealthConfig) error {
 	log := r.Log.WithValues("EtcdadmCluster", klog.KObj(etcdCluster))
-	if len(etcdCluster.Status.Endpoints) == 0 {
+
+	desiredReplicas := int(*etcdCluster.Spec.Replicas)
```

Comment:
can you check if replicas is nil before de-referencing it just so we don't run into any nil pointers